src/app/wallet/address-book/address-book.component.ts

```diff
@@ -1,15 +1,159 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import { Log } from 'ng2-logger'
+import { AddressService } from '../shared/address.service';
+import { Address } from '../../core/rpc/models/address.model';
+import { AddressCount } from '../../core/rpc/models/address-count.model';
+import { AddressType } from '../../core/rpc/models/address-type.enum';
+import {Subscription} from 'rxjs/Subscription';
+
 
 @Component({
   selector: 'app-address-book',
   templateUrl: './address-book.component.html',
-  styleUrls: ['./address-book.component.scss']
+  styleUrls: ['./address-book.component.scss'],
+  providers: [AddressService]
 })
-export class AddressBookComponent implements OnInit {
+export class AddressBookComponent implements OnInit, OnDestroy {
+
+  log: any = Log.create('address-book.component');
+
+  addressesSubscription: Subscription;
+  addressCountSubscription: Subscription;
+
+  addresses: Address[] = [];
+  addressCount: AddressCount = new AddressCount();
 
-  constructor() { }
+  // addressbook pagination state
+  currentPage: number = 1; // 1 -> first page
+  addressType: AddressType = AddressType.total;
+
+  // address-table configuration
+  itemsPerPage: number = 9; // TODO: move this to external config
+  displayHeader: boolean = false;
+  displayInternalHeader: boolean = true;
+  displayPagination: boolean = true;
+  displayLabel: boolean = true;
+  displayAddress: boolean = true;
+  displayPublicKey: boolean = false;
+  displayType: boolean = true;
+  displayPurpose: boolean = true;
+  displayToolsMenu: boolean = true;
+  displayQrMenu: boolean = true;
+
+  constructor(private addressService: AddressService) {
+  }
 
   ngOnInit() {
+
+    // update the addressCount
+    this.addressCountSubscription = this.updateAddressCount();
+
+    // get initial addresses using default settings
+    this.addressesSubscription = this.addressService
+      .findAddresses(this.getPaginationParams(this.addressType, this.currentPage))
```

Comment:
If we want to update the address table, we must re-call this function..
Do we inject a service in this component to tell us when we must update or do we move this higher up?